flight/targets/DTFc/fw/pios_board.c

```diff
@@ -0,0 +1,398 @@
+	HwSharedDSMxModeOptions hw_DSMxMode;
+	HwDTFcDSMxModeGet(&hw_DSMxMode);
+
+	/* Configure the rcvr port */
+	HwSharedPortTypesOptions hw_rcvrport;
+	HwDTFcRcvrPortGet(&hw_rcvrport);
+	PIOS_HAL_ConfigurePort(hw_rcvrport, // port_type
+						&pios_uart3_usart_cfg, // usart_port_cfg
+						&pios_usart_com_driver, // com_driver
+						NULL, // i2c_id
+						NULL, // i2c_cfg
+						&pios_ppm_cfg, // ppm_cfg
+						NULL, // pwm_cfg
+						PIOS_LED_ALARM, // led_id
+						&pios_uart3_dsm_cfg, // dsm_cfg
+						hw_DSMxMode, // dsm_mode
+						NULL); // sbus_cfg
+	
+	/* Configure Uart1 */
+	HwSharedPortTypesOptions hw_uart1;
+	HwDTFcUart1Get(&hw_uart1);
+	PIOS_HAL_ConfigurePort(hw_uart1, // port_type
+						&pios_uart1_usart_cfg, // usart_port_cfg
+						&pios_usart_com_driver, // com_driver
+						NULL, // i2c_id
+						NULL, // i2c_cfg
+						NULL, // ppm_cfg
+						NULL, // pwm_cfg
+						PIOS_LED_ALARM, // led_id
+						&pios_uart1_dsm_cfg, // dsm_cfg
+						hw_DSMxMode, // dsm_mode
+						NULL); // sbus_cfg
+	
+	/* Configure Uart2 */
+	HwSharedPortTypesOptions hw_uart2;
+	HwDTFcUart2Get(&hw_uart2);
+	PIOS_HAL_ConfigurePort(hw_uart2, // port_type
+						&pios_uart2_usart_cfg, // usart_port_cfg
+						&pios_usart_com_driver, // com_driver
+						NULL, // i2c_id
+						NULL, // i2c_cfg
+						NULL, // ppm_cfg
+						NULL, // pwm_cfg
+						PIOS_LED_ALARM, // led_id
+						&pios_uart2_dsm_cfg, // dsm_cfg
+						hw_DSMxMode, // dsm_mode
+						NULL); // sbus_cfg
+
+#if defined(PIOS_INCLUDE_GCSRCVR)
+	GCSReceiverInitialize();
+	uintptr_t pios_gcsrcvr_id;
+	PIOS_GCSRCVR_Init(&pios_gcsrcvr_id);
+	uintptr_t pios_gcsrcvr_rcvr_id;
+	if (PIOS_RCVR_Init(&pios_gcsrcvr_rcvr_id, &pios_gcsrcvr_rcvr_driver, pios_gcsrcvr_id)) {
+		PIOS_Assert(0);
+	}
+	pios_rcvr_group_map[MANUALCONTROLSETTINGS_CHANNELGROUPS_GCS] = pios_gcsrcvr_rcvr_id;
+#endif	/* PIOS_INCLUDE_GCSRCVR */
+
+#ifndef PIOS_DEBUG_ENABLE_DEBUG_PINS
+#ifdef PIOS_INCLUDE_SERVO
+	PIOS_Servo_Init(&pios_servo_cfg);
+#endif
+#else
+	PIOS_DEBUG_Init(&pios_tim_servo_all_channels, NELEMENTS(pios_tim_servo_all_channels));
+#endif
+
+#if defined(PIOS_INCLUDE_ADC)
+	uint32_t internal_adc_id;
+	if (PIOS_INTERNAL_ADC_Init(&internal_adc_id, &internal_adc_cfg) < 0)
+		PIOS_HAL_Panic(PIOS_LED_ALARM, PIOS_HAL_PANIC_ADC);
+	if (PIOS_ADC_Init(&pios_internal_adc_id, &pios_internal_adc_driver, internal_adc_id) < 0)
+		PIOS_HAL_Panic(PIOS_LED_ALARM, PIOS_HAL_PANIC_ADC);
+	
+	// Enable battery monitor module.
+	ModuleSettingsData modulesettings;
+	ModuleSettingsGet(&modulesettings);
+	modulesettings.AdminState[MODULESETTINGS_ADMINSTATE_BATTERY] = MODULESETTINGS_ADMINSTATE_ENABLED;
+	ModuleSettingsSet(&modulesettings);
+	
+	// Set voltage/current calibration values.
+	FlightBatterySettingsInitialize();
+	FlightBatterySettingsData batterysettings;
+	FlightBatterySettingsGet(&batterysettings);
+	batterysettings.SensorCalibrationFactor[FLIGHTBATTERYSETTINGS_SENSORCALIBRATIONFACTOR_CURRENT] = (float)DTFC_CURRENT_CALIBRATION_VALUE;
```

Comment:
Would be good to let GCS figure out the proper default ratio and set it in the setup wizard or with a button on the battery tab or something.